src/notebook.c

```diff
@@ -455,60 +456,104 @@ static void on_close_documents_right_activate(GtkMenuItem *menuitem, GeanyDocume
 }
 
 
+// simple implementation (vs. close all which doesn't close documents if cancelled)
+static void close_folder_action(GeanyDocument *cur_doc, gboolean other_folders)
+{
+	if (cur_doc == NULL)
+		cur_doc = document_get_current();
+	if (!cur_doc->real_path)
+		return;
+
+	gchar *dir = g_dirname(cur_doc->real_path);
```

Comment:
Should probably use `g_path_get_dirname()` since `g_dirname()` is deprecated.

Reply:
Thanks, fixed.